string objective_name
int8 objective_state
ddrp/Region initial_region
ddrp/Region target_region
